webapp/store/actions/clientActions.js

```diff
@@ -10,35 +10,47 @@ function setClients(clients) {
   };
 }
 
-export function setCurrentClient(clientInfo) {
+function setCurrentClient(clientInfo) {
   const { id, chain = 'bitcoin' } = clientInfo;
   // set the client info for the global client
   if (id) client.setClientInfo(id, chain);
   return {
-    type: SET_DEFAULT_CLIENT,
+    type: SET_CURRENT_CLIENT,
     payload: clientInfo
   };
 }
 
-export function getClients() {
+function resetClient() {
+  return {
+    type: 'RESET_STATE'
+  };
+}
+
+function getClients() {
   return async dispatch => {
     const clients = await client.getClients();
     dispatch(setClients(clients));
   };
 }
 
-export function getCurrentClient() {
+function getCurrentClient() {
   return async dispatch => {
     const currentClient = await client.getDefault();
     dispatch(setCurrentClient(currentClient));
   };
 }
 
-export function hydrateClients() {
-  return async dispatch => {
+function hydrateClients() {
+  return async (dispatch, getState) => {
     try {
       await dispatch(getClients());
-      await dispatch(getCurrentClient());
+      let currentClient = getState().clients.currentClient;
+      if (!currentClient.id) {
```

Comment:
What is the edge case that you are handling here?

Reply:
This is when you don't have a client store hydrated from localStorage and so we need to find out what to default our client to from the server. 

Actually, now that you mention it, it might be better to update `getCurrentClient` to `getDefaultClient` since that's all it's doing now.

Reply:
Is this getting updated?

Reply:
Yes, fixed.